tracer/src/Datadog.Trace/AppSec/Security.cs

```diff
@@ -201,27 +202,71 @@ private static void TryAddEndPoint(Span span)
             }
         }
 
+        private static bool AreArchitectureAndOsSupported()
+        {
+            var frameworkDescription = FrameworkDescription.Instance;
+            var osSupported = false;
+            var supportedOs = new[]
+            {
+                OSPlatform.Linux,
+                OSPlatform.MacOS,
+                OSPlatform.Windows
+            };
+            if (supportedOs.Contains(frameworkDescription.OSPlatform))
+            {
+                osSupported = true;
+            }
+
+            var archSupported = false;
+            var supportedArchs = new[]
+            {
+                ProcessArchitecture.Arm,
+                ProcessArchitecture.X64,
+                ProcessArchitecture.X86
+            };
+            if (supportedArchs.Contains(frameworkDescription.ProcessArchitecture))
+            {
+                archSupported = true;
+            }
```

Comment:
For performance reasons, we should avoid allocating these array on every call (moving them to a static field, for example). And since they only contain 3 values each, we could go further and avoid the arrays and the LINQ `Contains()` entirely and instead do the 3 comparisons.

You should also consider comparing `RuntimeInformation.ProcessArchitecture` values directly to `System.Runtime.InteropServices.Architecture` enum values, instead of comparing strings. Our `FrameworkDescription` was designed mostly for outputting values to logging, so everything is a string 😅

Combining all of the above, this could be written something like this:
```suggestion
            var processArch = RuntimeInformation.ProcessArchitecture;

            var archSupported = processArch == Architecture.X86 ||
                                processArch == Architecture.X64 ||
                                processArch == Architecture.Arm64;
```

Reply:
Same comment for `supportedOs` above.

Reply:
Good point thanks. But as this is not part of this PR (this method has just been moved further down), I suggest I do another small PR to change it, cause that involves creating partial classes to handle both Core and Framework etc ( like framework description used to).

Reply:
This is only called in the constructor of `Security()` right, which is called once on app startup. So I think that storing the values in a static could actually be undesirable (as they won't be needed again), although skipping LINQ could be useful..

Reply:
Good point, it's run only once, I removed LINQ and changed Arm in 14d88798742d6e3e2b9f118bb980b48882ad0bdf, just realized we actually don't support it yet!, once we update the nuget package arm64 should then be supported but for now I'll remove it altogether.